vnext/ReactUWP/Views/MenuFlyoutViewManager.cpp

```diff
@@ -0,0 +1,436 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+
+#include "MenuFlyoutViewManager.h"
+#include "MenuFlyoutItemViewManager.h"
+
+#include <Views/ShadowNodeBase.h>
+#include "TouchEventHandler.h"
+#include "ViewPanel.h"
+#include <Utils/ValueUtils.h>
+
+#include <Modules/NativeUIManager.h>
+#include <Utils/PropertyHandlerUtils.h>
+#include <winrt/Windows.UI.Xaml.Controls.Primitives.h>
+
+namespace winrt {
+  using namespace Windows::UI::Xaml::Controls::Primitives;
+  using namespace Windows::UI::Xaml::Interop;
+}
+
+static const std::unordered_map<std::string, winrt::FlyoutPlacementMode>
+    placementModeMinVersion = {{"top", winrt::FlyoutPlacementMode::Top},
+                               {"bottom", winrt::FlyoutPlacementMode::Bottom},
+                               {"left", winrt::FlyoutPlacementMode::Left},
+                               {"right", winrt::FlyoutPlacementMode::Right},
+                               {"full", winrt::FlyoutPlacementMode::Full}};
+
+static const std::unordered_map<std::string, winrt::FlyoutPlacementMode>
+    placementModeRS5 = {{"top", winrt::FlyoutPlacementMode::Top},
+                        {"bottom", winrt::FlyoutPlacementMode::Bottom},
+                        {"left", winrt::FlyoutPlacementMode::Left},
+                        {"right", winrt::FlyoutPlacementMode::Right},
+                        {"full", winrt::FlyoutPlacementMode::Full},
+                        {"top-edge-aligned-left",
+                         winrt::FlyoutPlacementMode::TopEdgeAlignedLeft},
+                        {"top-edge-aligned-right",
+                         winrt::FlyoutPlacementMode::TopEdgeAlignedRight},
+                        {"bottom-edge-aligned-left",
+                         winrt::FlyoutPlacementMode::BottomEdgeAlignedLeft},
+                        {"bottom-edge-aligned-right",
+                         winrt::FlyoutPlacementMode::BottomEdgeAlignedRight},
+                        {"left-edge-aligned-top",
+                         winrt::FlyoutPlacementMode::LeftEdgeAlignedTop},
+                        {"left-edge-aligned-bottom",
+                         winrt::FlyoutPlacementMode::LeftEdgeAlignedBottom},
+                        {"right-edge-aligned-top",
+                         winrt::FlyoutPlacementMode::RightEdgeAlignedTop},
+                        {"right-edge-aligned-bottom",
+                         winrt::FlyoutPlacementMode::RightEdgeAlignedBottom}};
+
+template <>
+struct json_type_traits<winrt::FlyoutPlacementMode> {
+  static winrt::FlyoutPlacementMode parseJson(const folly::dynamic &json) {
+    auto placementMode = !!(winrt::Flyout().try_as<winrt::IFlyoutBase5>())
+        ? placementModeRS5
+        : placementModeMinVersion;
+    auto iter = placementMode.find(json.asString());
+
+    if (iter != placementMode.end()) {
+      return iter->second;
+    }
+
+    return winrt::FlyoutPlacementMode::Right;
+  }
+};
```

Comment:
Can you try to share the code with FlyoutViewManager? I see you can duplicated the code